app/src/main/java/com/ranveeraggarwal/letrack/activities/AddPersonActivity.java

```diff
@@ -37,24 +48,48 @@
     String selectedFrequency;
     String selectedStartDate;
 
+    int REQUEST_CONTACTS = 1;
+    String TAG = "AddPersonActivity";
+    boolean hasContactPermission = false;
+
     DatabaseAdapter databaseAdapter;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
         setContentView(R.layout.activity_add_person);
+        if (ContextCompat.checkSelfPermission(
+                this, Manifest.permission.READ_CONTACTS) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, new String[] {Manifest.permission.READ_CONTACTS}, REQUEST_CONTACTS);
+        } else {
+            hasContactPermission = true;
+        }
```

Comment:
If I'm not wrong, this will check for permissions every time you open the activity. Say I don't want to use this feature, I don't want to give contact permissions and I still want to use the app. So, this part should be moved to the `onClickListener` function instead.